pennylane/pulse/rydberg.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the classes/functions needed to simulate the evolution of ensembles of
+individual (trapped) rydberg atoms under the excitation of several laser fields."""
+import numpy as np
+
+import pennylane as qml
+from pennylane.ops import SProd, Sum
+
+from .parametrized_hamiltonian import ParametrizedHamiltonian
+
+
+class RydbergEnsemble:
+    """Class representing the interaction of an ensemble of Rydberg atoms.
+
+    Args:
+        qubit_positions (list): list of coordinates of each atom in the ensemble
+        wires (list): List of wires for each atom in the ensemble. If ``None``, the wire values
+            correspond to the index of the atom in the list of qubit positions. Defaults to ``None``.
+
+    Returns:
+        RydbergEnsemble: class representing an ensemble of Rydberg atoms
+    """
+
+    def __init__(self, qubit_positions: list, wires: list = None) -> None:
+        self.qubit_positions = qubit_positions
```

Comment:
I guess right now the positions are in meters, right? (Should be noted in the docstring)

How well does the solver deal with large/small orders of magnitude? Realistic values for these numbers are on the order of 1e-6 meters. Further down I also asked questions about hbar and the scaling constant on the Rydberg interaction, which are quite a bit more extreme. Can we safely use numbers like that, or should be be trying to normalize things to keep them within some range?

Reply:
Good point!

I reckon that this can be "normalized" by the timescales. E.g. in the transmon qubits everything is in GHz, so the time scale is chosen to be in nanoseconds such that overall we can avoid very large or very small numbers for numerical accuracy

Reply:
Isnt the distance different for every pair? And should be inferred from the positions of the atoms on the grid?

Reply:
I thought atoms could only be placed in a squared grid, but maybe it is better to give the user more freedom.

I removed the `distance` argument and mentioned in the doc that the coordinates are in micrometers.